### ISSUE TITLE
Invalid parameter 'discard'

### ISSUE THREAD

**Issue by joshuacox** [mitchellh/packer#2376]
Any reason why discard is an invalid parameter? 

```
2015/07/01 17:01:08 packer-builder-qemu: 2015/07/01 17:01:08 Qemu Builder has no floppy files, not attaching a floppy.
2015/07/01 17:01:08 packer-builder-qemu: 2015/07/01 17:01:08 Executing /usr/bin/qemu-system-x86_64: []string{"-machine", "type=pc,accel=kvm", "-netdev", "user,id=user.0,hostfwd=tcp::3213-:22", "-drive", "file=buids/dacksible-ubuntu-1404-amd64-qemu/dacksible-ubuntu-1404-amd64,if=virtio,cache=writeback,discard=ignore", "-cdrom", "/behemoth/git/dacksible/packer_cache/858ac1e85e6189fb336e35825f1ad05bef9e7a5b2e9d191cfee70fa3a9d8527.iso", "-boot", "once=d", "-name", "dacksible-ubuntu-1404-amd64", "-device", "virtio-net,netdev=user.0", "-m", "512M", "-vnc", "0.0.0.0:47"}                                                                  
2015/07/01 17:01:08 packer-builder-qemu: 2015/07/01 17:01:08 Started Qemu. Pid: 25386
2015/07/01 17:01:08 packer-builder-qemu: 2015/07/01 17:01:08 Qemu stderr: qemu-system-x86_64: -drive file=builds/dacksible-ubuntu-1404-amd64-qemu/dacksible-ubuntu-1404-amd64,if=virtio,cache=writeback,discard=ignore: Invalid parameter 'discard' 
```

uname -r
```
uname -r
3.2.0-4-amd64
```

**Comment by vtolstov** [comment]
What qemu version?
02 июля 2015 г. 3:17 пользователь "Josh Cox" <notifications@github.com>
написал:

**Comment by AlanDewar** [comment]
I'm seeing the same issue "discard=ignore: Invalid parameter 'discard'" with:

   QEMU emulator version 1.1.2 (Debian 1.1.2+dfsg-6a+deb7u8)

but have worked around it using:

    "qemuargs": [ "-drive", "file=output-qemu/debian-vm,if=virtio,cache=writeback" ],

**Comment by AlanDewar** [comment]
A quick follow-on from my previous reply.

I've just upgraded qemu to:

  QEMU emulator version 2.1.2 (Debian 1:2.1+dfsg-12~bpo70+1),

and the  "discard=ignore: Invalid parameter 'discard'" has gone away.

**Comment by vtolstov** [comment]
I think we need to check qemu version and enable/disable this feature based on version. Please don't close issue, so i know about it and not forget =).

**Comment by joshuacox** [comment]
sorry I forgot all about this issue, I built on another machine and moved on, however, the machine that had the issue was debian wheezy, and according to this that is now 1.1.2 :

https://packages.debian.org/wheezy/qemu-kvm

the machine I completed the build on was jessie, so assumedly this is fixed jessie forward.  I'll leave it open and let you close when you want, please let me know if I can gather any more info.

**Comment by GonZo** [comment]
:+1: 
Centos 6
QEMU PC emulator version 0.12.1 (qemu-kvm-0.12.1.2-2.448.el6_6)

**Comment by markpeek** [comment]
Fixed with #2422.
